marmopy/intent.py

```diff
@@ -102,5 +110,48 @@ def to_json(self):
             "signature": self.signature
         }
 
-    def relay(self, relayer):
-        return requests.post(relayer, json=self.to_json())
+    def status(self, provider = None):
+        if not provider:
+            provider = global_provider()
+            assert provider
+
+        w3 = provider.web3
+        contract = w3.eth.contract(abi = json.loads(wallet_abi), address = self.wallet.address)
+
+        try:
+            block = contract.call().relayedAt(Web3.toBytes(hexstr=self.id))
+        except BadFunctionCallOutput:
+            return StatusReceipt(StatusCode.pending)
+
+        if block != 0:
```

Comment:
si el block es 0 puede ser por que esta mal el id del intent o la wallet no esta deployada, no deberia devolver un error en lugar de un pending ?

Reply:
If the wallet is not deployed, it just meant that the relayer didn't reveal the wallet yet, but the Intent is still valid.